### DISCUSSION TITLE
Extend multiple classes

### DISCUSSION THREAD

**Discussion by adriangb** [PyO3/pyo3#3143] [Questions]
Is there any way to do the equivalent of:

```python
class MyBaseException(BaseException):
    pass

class MyException(MyBaseException, Exception):
    pass
```

It seems like you [can't have more than one base in `extends`](https://pyo3.rs/v0.18.3/class.html?highlight=extends#inheritance)

**Comment by davidhewitt** [comment]
In principle we could support multiple base types, however as far as I know the Python bases need to have "compatible layouts" and that's not currently something designed for in our `PyCell` type.

I think if we supported it, it should be opt-in; `nanobind` chose to drop multiple inheritance support to simplify compared to `pybind11`.

**Reply by phoerious** [reply]
I would be interested in this as well. I have multiple mixin bases and have to duplicate all my subclass implementations and doc strings, because I can inherit only from a single class. This is made worse by the fact that I cannot use macros inside #[pymethods], so I actually have to copy-paste all my pymethods to all subclasses.

**Comment by OutSquareCapital** [comment]
Is there any way to do this with the C API?

**Reply by Person-93** [reply]
From the docs: https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_base

> multiple inheritance require dynamically creating a type object by calling the metatype.
